tools/bro-to-vast/bro-to-vast.cpp

```diff
@@ -152,16 +152,20 @@ broker::data to_broker(const vast::data& data) {
 }
 
 // Constructs a result event for Bro from Broker data.
-broker::bro::Event make_result_event(std::string name, broker::data x) {
+broker::bro::Event make_result_event(std::string query_id, broker::data x) {
   broker::vector args(2);
-  args[0] = std::move(name);
+  args[0] = std::move(query_id);
   args[1] = std::move(x);
   return {"VAST::result", std::move(args)};
 }
 
 // Constructs a result event for Bro from a VAST event.
-broker::bro::Event make_result_event(const vast::event& x) {
-  return make_result_event(x.type().name(), to_broker(x.data()));
+broker::bro::Event make_result_event(std::string query_id,
+                                     const vast::event& x) {
+  broker::vector xs(2);
```

Comment:
Why not initialize the vector via xs{broker::data{x.type().name()}, to_broker(x.data())}`?

Reply:
Because it's my understanding that going through the `initializer_list` ctor incurs extra copies.

Reply:
And here's the proof:
```cpp
#include <iostream>
#include <vector>

using namespace std;

struct foo {
  foo() {
    cout << "ctor" << endl;
  }

  foo(const foo&) {
    cout << "copy" << endl;
  }

  foo(foo&&) {
    cout << "move" << endl;
  }

  foo& operator=(const foo&) {
    cout << "copy=" << endl;
    return *this;
  }

  foo& operator=(foo&&) {
    cout << "move=" << endl;
    return *this;
  }
};

int main() {
  vector<foo> xs(1);
  cout << "------------" << endl;
  xs[0] = foo{};
  cout << "------------" << endl;
  vector<foo> ys{foo{}};
}
```
Prints:
```
ctor
------------
ctor
move=
------------
ctor
copy
```